File: frontend/shop_kingkong/src/service/CartService.js
import axios from "axios";

const CartService = {
  getAll() {
    return axios.get("cart");
  },
  getCartById(productId) {
    return axios.get(`cart/${cartId}`);
  },
  createCart(productRequest) {
    return axios.get(`cart/create/${cartRequest}`);
  },
  updateCart(cartId, cartRequest) {
    return axios.get(`cart/update/${cartId}/${cartRequest}`);
  },
  deleteCart(cartId) {
    return axios.get(`cart/delete/${cartId}`);
  },
};

File: backend/src/main/java/bag/service/cart/CartService.java
package bag.service.cart;

import bag.modal.dto.CartDto;
import bag.modal.entity.Cart;
import bag.modal.request.CartRequest;

import java.util.List;

public interface CartService {
    List<CartDto> getAllCarts();

    CartDto getCartById(int id);

    CartDto createCart(CartRequest request);

    CartDto updateCart(CartRequest request, int id);

    void deleteCart(int id);

}
